epilepsy12/forms_folder/case_form.py

```diff
@@ -79,19 +84,68 @@ class CaseForm(forms.ModelForm):
 
     def __init__(self, *args, **kwargs) -> None:
-        
+        self.organisation_id = kwargs.pop(
+            "organisation_id", None
+        )  # This is the organisation_id
+
+        # set a flag to check if this is Jersey
+        self.is_jersey = (
+            Organisation.objects.get(
+                id=self.organisation_id
+            ).country.boundary_identifier
+            == "JEY"
+        )
         super(CaseForm, self).__init__(*args, **kwargs)
+        self.existing_nhs_number = self.instance.nhs_number
         self.fields["ethnicity"].widget.attrs.update({"class": "ui rcpch dropdown"})
+        if self.is_jersey:
+            # this is Jersey - hide the NHS number field
+            self.fields["nhs_number"].widget = forms.HiddenInput()
+            self.fields["nhs_number"].required = False
+            self.fields["unique_reference_number"].required = True
+            self.fields["unique_reference_number"].initial = (
+                self.instance.unique_reference_number
+            )
+        else:
+            # this is England or Wales - hide the URN field
+            self.fields["unique_reference_number"].widget = forms.HiddenInput()
+            self.fields["unique_reference_number"].required = False
+            self.fields["nhs_number"].required = True
+            self.fields["nhs_number"].initial = self.instance.nhs_number
 
-        self.existing_nhs_number = self.instance.nhs_number
-        
         # Check if DEBUG is True and set the initial value conditionally
         if settings.DEBUG:
-            self.fields['first_name'].initial  = 'Bob'
-            self.fields['surname'].initial = 'Dylan'
-            self.fields['date_of_birth'].initial = date(randint(2005, 2021), randint(1, 12), randint(1, 28))
-            self.fields['postcode'].initial = return_random_postcode(country_boundary_identifier='E01000001')
-            self.fields['nhs_number'].initial = nhs_number.generate()[0]
+            self.fields["first_name"].initial = "Bob"
+            self.fields["surname"].initial = "Dylan"
+            self.fields["date_of_birth"].initial = date(
+                randint(2005, 2021), randint(1, 12), randint(1, 28)
+            )
+            # set a random postcode if DEBUG is True: E01000001 is the boundary identifier for England but if is_jersey is True
+            # then a Jersey postcode will be returned instead of an English/Welsh postcode
+            self.fields["postcode"].initial = return_random_postcode(
+                country_boundary_identifier=Organisation.objects.get(
+                    id=self.organisation_id
+                ).country.boundary_identifier,
+                is_jersey=self.is_jersey,
+            )
+            if self.is_jersey:
+                # this is Jersey
+                if self.instance.unique_reference_number is None:
+                    # this is a new form - create a new URN that is unique
+                    while True:
```

Comment:
It is gated behind a DEBUG flag but I'm always a bit scared at having a `while True` buried within nested code - it's a risk that this could run unexpectedly and crash the app.

Do we need it? We don't bother with it for NHS numbers below. We could also just use a UUID

Reply:
good point. will remove